src/SFA.DAS.AANHub.Application/Employers/Commands/CreateEmployerMemberCommandValidator.cs

```diff
@@ -7,25 +7,39 @@ namespace SFA.DAS.AANHub.Application.Employers.Commands
 {
     public class CreateEmployerMemberCommandValidator : AbstractValidator<CreateEmployerMemberCommand>
     {
-        public CreateEmployerMemberCommandValidator(IRegionsReadRepository regionsReadRepository, IMembersReadRepository membersReadRepository, IEmployersReadRepository employersReadRepository)
+        private const string EmployerPairAlreadyCreatedErrorMessage = "UserId and AccountId pair already exist";
+
+        private readonly IEmployersReadRepository _employersReadRepository;
+
+        public CreateEmployerMemberCommandValidator(IRegionsReadRepository regionsReadRepository, IMembersReadRepository membersReadRepository,
+            IEmployersReadRepository employersReadRepository)
         {
+            _employersReadRepository = employersReadRepository;
+
             Include(new CreateMemberCommandBaseValidator(regionsReadRepository));
             Include(new RequestedByMemberIdValidator(membersReadRepository));
             RuleFor(c => c.UserId)
                 .NotEmpty();
+
             RuleFor(c => c.Organisation)
                 .NotEmpty()
                 .MaximumLength(250);
+
             RuleFor(c => c.AccountId)
-                .NotEmpty();
-            RuleFor(c => c)
                 .NotEmpty()
-                .MustAsync(async (command, cancellation) =>
-                {
-                    var result = await employersReadRepository.GetEmployerByAccountIdAndUserId(command.AccountId, command.UserId);
-                    return result == null;
-                })
-                .WithMessage("UserId and AccountId pair already exist");
+                .MustAsync(async (command, _, _) => await CheckAccountIdAndUserIdPair(command.AccountId, command.UserId))
```

Comment:
Perhaps going forward unrelated impactful changes should be covered in a separate task/PR as QAs will miss these unless they are told.

Reply:
Yes, myself Olaniyi agreed that the fix for CSP-556 would be part of this ticket to unblock the rest of the backlog, and that situations like this would not be regular thing going forward.

@OlaniyiOshinusi is aware of the additional test in this ticket.

Reply:
That's good then. Thanks